backend/director/agents/base.py

```
@@ -41,6 +41,20 @@ def get_parameters(self):
             )
         return parameters
 
+    def _check_supported_llm(self):
+        """Check if supported_llm is present and validate LLM."""
+        if hasattr(self, "supported_llm") and hasattr(self, "llm"):
+            if self.llm.llm_type not in self.supported_llm:
```

Ruff (0.8.0)

Use a single `if` statement instead of nested `if` statements

(SIM102)